incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -3811,7 +3780,6 @@ private void cleanupEnd(
                 long traceId,
                 long authorization)
             {
-                doPublishNetEnd(traceId, authorization, EMPTY_OCTETS);
```

Comment:
Suggest that the publish expiration should be cleaned up implicitly when the stream is closed, i.e. called from `doPublishAppEnd` and `doPublishAppAbort`, agree?

Then `cleanupEnd` reduces to `doPublishAppEnd` and calls to `cleanupEnd` can be replaced by `doPublishAppEnd` and `cleanupEnd` can be removed.